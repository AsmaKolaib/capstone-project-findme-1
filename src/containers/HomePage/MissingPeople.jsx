import React from 'react';
import MissingPersonCard from '../../components/MissingPersonCard';

const HomePage = ({ missingPeople }) => (
  <div className="">
    <h2 className="font-normal tracking-wider text-4xl mb-10 mx-4">
      {missingPeople}
    </h2>
    <div className="flex flex-col md:flex-row lg:flex-row xl:flex-row sm:flex-col sm:items-center">
      <MissingPersonCard userName="John Doe" lostSince="7-7-2017" />
      <MissingPersonCard userName="John Doe" lostSince="7-7-2017" />
      <MissingPersonCard userName="John Doe" lostSince="7-7-2017" />
      <MissingPersonCard userName="John Doe" lostSince="7-7-2017" />
    </div>
    <p className="mx-4 my-10 text-right text-blue-500">
      <a href="#">
        More Missing People &rarr;{/** this utf for -> [right arrow icon] */}
      </a>
    </p>
  </div>
);

export default HomePage;
