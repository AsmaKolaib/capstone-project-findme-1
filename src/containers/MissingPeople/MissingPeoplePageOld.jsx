import React, { useState } from 'react';
import MissingPersonCard from '../../components/MissingPersonCard';
import Api from './api';

const MissingPeoplePage = () => {
  const [data, setData] = useState(Api);
  // console.log(data);

  return (
    <div className="">
      <h2 className="font-normal tracking-wider text-4xl mb-10 mx-4">
        Missing People
      </h2>
      <div className="flex flex-wrap">
        {data.length === 0 ? (
          <h2 className="flex-auto text-center">not found results</h2>
        ) : (
          data.map((item, index) => (
            <MissingPersonCard key={item.id} cardInfo={item} i={index} />
          ))
        )}
      </div>
      <div className="">
        <button className="mx-auto my-10 py-2 px-4 align-center text-blue-500 border-solid border-2 border-blue-500 rounded-3xl hover:text-blue-600 hover:border-blue-600 block font-light">
          <a href="#">
            <span className="">â¬‡ Show More</span>
          </a>
        </button>
      </div>
    </div>
  );
};

export default MissingPeoplePage;
