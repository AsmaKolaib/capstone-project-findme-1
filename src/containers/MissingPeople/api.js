const api = [
  {
    id: 1,
    userName: 'John Doe',
    lostSince: '6-6-2016',
    img: require('./images/profile-svg.svg'),
    lostFamily: true,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 120,
  },
  {
    id: 2,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/image.jpg'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 220,
  },
  {
    id: 3,
    userName: 'John Doe',
    lostSince: '8-8-2018',
    img: require('./images/image2.jpg'),
    lostFamily: true,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 240,
  },
  {
    id: 4,
    userName: 'John Doe',
    lostSince: '9-9-2019',
    img: require('./images/image3.png'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 260,
  },
  {
    id: 5,
    userName: 'John Doe',
    lostSince: '10-10-2020',
    img: require('./images/image4.jpg'),
    lostFamily: true,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 280,
  },
  {
    id: 6,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/image5.jpg'),
    lostFamily: true,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 300,
  },
  {
    id: 7,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/image6.png'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 150,
  },
  {
    id: 8,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 250,
  },
  {
    id: 9,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: true,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 80,
  },
  {
    id: 10,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 83,
  },
  {
    id: 11,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: true,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 35,
  },
  {
    id: 12,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 24,
  },

  {
    id: 13,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: true,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 80,
  },
  {
    id: 14,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 83,
  },
  {
    id: 15,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: true,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 35,
  },
  {
    id: 16,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 24,
  },
  {
    id: 17,
    userName: 'John Doe',
    lostSince: '7-7-2017',
    img: require('./images/profile-svg.svg'),
    lostFamily: false,
    description:
      'This website has helped me find my son, I will always be thankful for that.',
    age: 17,
    location: "Sana'a",
    views: 24,
  },
];
export default api;
