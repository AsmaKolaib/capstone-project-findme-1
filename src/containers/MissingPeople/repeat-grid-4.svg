<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="70" height="62" viewBox="0 0 70 62">
  <defs>
    <clipPath id="clip-path">
      <rect width="70" height="62" fill="none"/>
    </clipPath>
  </defs>
  <g id="Repeat_Grid_4" data-name="Repeat Grid 4" opacity="0.3" clip-path="url(#clip-path)">
    <g transform="translate(1066 -2147)">
      <rect id="Rectangle_31" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1082 -2147)">
      <rect id="Rectangle_31-2" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1098 -2147)">
      <rect id="Rectangle_31-3" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1114 -2147)">
      <rect id="Rectangle_31-4" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1130 -2147)">
      <rect id="Rectangle_31-5" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1066 -2133)">
      <rect id="Rectangle_31-6" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1082 -2133)">
      <rect id="Rectangle_31-7" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1098 -2133)">
      <rect id="Rectangle_31-8" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1114 -2133)">
      <rect id="Rectangle_31-9" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1130 -2133)">
      <rect id="Rectangle_31-10" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1066 -2119)">
      <rect id="Rectangle_31-11" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1082 -2119)">
      <rect id="Rectangle_31-12" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1098 -2119)">
      <rect id="Rectangle_31-13" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1114 -2119)">
      <rect id="Rectangle_31-14" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1130 -2119)">
      <rect id="Rectangle_31-15" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1066 -2105)">
      <rect id="Rectangle_31-16" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1082 -2105)">
      <rect id="Rectangle_31-17" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1098 -2105)">
      <rect id="Rectangle_31-18" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1114 -2105)">
      <rect id="Rectangle_31-19" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1130 -2105)">
      <rect id="Rectangle_31-20" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1066 -2091)">
      <rect id="Rectangle_31-21" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1082 -2091)">
      <rect id="Rectangle_31-22" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1098 -2091)">
      <rect id="Rectangle_31-23" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1114 -2091)">
      <rect id="Rectangle_31-24" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
    <g transform="translate(1130 -2091)">
      <rect id="Rectangle_31-25" data-name="Rectangle 31" width="6" height="6" rx="3" transform="translate(-1066 2147)" fill="#138def"/>
    </g>
  </g>
</svg>
