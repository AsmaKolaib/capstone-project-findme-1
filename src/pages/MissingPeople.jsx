import React from 'react';

//import MissingPeoplePage from '../containers/MissingPeople/MissingPeoplePage';
import Search from '../components/Search/Search';

const MissingPeople = () => {
  return (
    <div className="App bg-findMe">
      <Search />
    </div>
  );
};

export default MissingPeople;
