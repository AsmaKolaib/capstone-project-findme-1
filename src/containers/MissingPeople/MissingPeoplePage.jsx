import React, { useState, useEffect, useRef } from 'react';
// import { CircleArrow as ScrollUpButton } from 'react-scroll-up-button';
import MissingPersonCard from '../../components/MissingPersonCard/MissingPersonCard';
import Model from 'react-modal'
import { ReactComponent as MyIcon } from './repeat-grid-4.svg';
import Search from '../../components/Search/Search'
import './style.css';
import NiceButton from '../../components/LoadingButton/NiceButton';
import '../../components/LoadingButton/style.css';
import useFirestore from '../../hooks/useFirestore'
import { Link, Route } from 'react-router-dom';
import { getSearchInfo } from '../../components/Search/searchInfo'
import {getCurrentDate} from '../../components/Search/helpers'
const MissingPeoplePage = ({searchInfo, searched}) => {
  const ITEMSTOSHOW = 8;
  const { docs } = useFirestore('images');
  const [data, setData] = useState([]); // maybe set the initial value later
  const [visible, setVisible] = useState(ITEMSTOSHOW);
  const [buttonText, setButtonText] = useState('⬇ Show More');
  const buttonTextRef = useRef('⬇ Show More');
  const [matched, setMatched] = useState([]);
  const [isSecondButtonLoading, setIsSecondButtonLoading] = useState(false);
  const [loadingSpeed, setLoadingSpeed] = useState(1);
  const [isOpen, setIsOpen] = useState(false);

  const dynamicSearch =(entered)=>{

    if (searchInfo.goneMissingOn===getCurrentDate()&& !entered.isLookingForFamily  )
     return docs.filter(name=> name.values.fristName.toLowerCase().includes(entered.missingName.toLowerCase()))
    else if (searchInfo.goneMissingOn===getCurrentDate()&& entered.isLookingForFamily)
     return docs.filter(name=> name.values.fristName.toLowerCase().includes(entered.missingName.toLowerCase())
     && name.values.isLookingFor)
     else if (searchInfo.goneMissingOn!==getCurrentDate()&& !entered.isLookingForFamily)
     return docs.filter(name=>{ return name.values.lostSince.toLowerCase().includes(entered.goneMissingOn.toLowerCase()) && 
      name.values.fristName.toLowerCase().includes(entered.missingName.toLowerCase())
      
    })

     else 
     return docs.filter(name=>{  name.values.lostSince.toLowerCase().includes(entered.goneMissingOn.toLowerCase()) && 
      name.values.fristName.toLowerCase().includes(entered.missingName.toLowerCase()&& name.values.isLookingFor)
      
    })
    
    
    }
  useEffect(() => {
    if (isSecondButtonLoading) {
      setTimeout(() => {
        setIsSecondButtonLoading(false);
      }, 1000 / loadingSpeed);
    }
  }, [isSecondButtonLoading, loadingSpeed]);

  useEffect(() => {
    setData(() => docs);
  }, [docs]);
  const getData =()=>{

  }
  
  const showMoreItems = () => {
    if (visible > docs.length) {
      // setButtonText(() => 'complete');
      buttonTextRef.current = 'Complete';
    }

    // setButtonText(() => '...loading');
    // buttonTextRef.current = '..loading';
    setVisible((prevValue) => prevValue + ITEMSTOSHOW);
    // buttonTextRef.current = '⬇ Show More';
    // setButtonText(() => '⬇ Show More');
  };

  const showMoreItemsClick = () => {
    // buttonTextRef.current = '..loading';
    setTimeout(() => {
      showMoreItems();
    }, 1000);
    clearTimeout();
    // buttonTextRef.current = '⬇ Show More';
  };

  return (
    <div>

      <MyIcon className="absolute" />
      <MyIcon className="absolute right-0 mt-64 " />

      <div className="flex flex-wrap ml-8 xl:justify-center">
        {data.length === 0 ? (
          <h2 className="flex-auto text-center">not found results</h2>
        ) : (
          searched?(
           // dynamicSearch(docs, searchInfo)
           /*matched = docs && docs
            .filter((item, index) => (
             item.values.fristName.toLowerCase().includes(searchInfo.missingName.toLowerCase())

            ))*/
            dynamicSearch(searchInfo)
            .slice(0, visible)
            .map((item, index) => (
              //searchFun(item, searchInfo)==='nameSearch'?(
             // item.values.fristName.toLowerCase()===searchInfo.missingName ?
              <Link key={item.id}
              to={`/missing_people_details/${item.id}`}
                >
              <MissingPersonCard key={item.id}  id={item.id} cardInfo={item} i={index} />
              </Link>


                 )     )
                  
            )
            : (
            docs && docs
            .slice(0, visible)
            .map((item, index) => (
              <Link key={item.id}
              to={`/missing_people_details/${item.id}`}
                >
              <MissingPersonCard key={item.id}  id={item.id} cardInfo={item} i={index} />
              </Link>
            ))
            
        ))}
      </div>
      {console.log(visible, data.length)}
      <div className=" ">
        {visible > docs.length || visible > dynamicSearch(searchInfo).length ?  (
          ''
        ) : (
          <NiceButton
            className="focus:outline-none mx-auto my-10 py-2 px-4 align-center text-blue-500 border-solid border-2 border-blue-500 rounded-3xl hover:text-blue-600 hover:border-blue-600 block font-light "
            isLoading={isSecondButtonLoading}
            onClick={() => {
              setIsSecondButtonLoading(true);
              showMoreItemsClick();
            }}
          >
            ⬇ Show More
          </NiceButton>
        )}
      </div>
    </div>
  );
};

// const removeOutline = () => {};

export default MissingPeoplePage;

// const scrollButtonStyle = {
//   position: 'fixed',
//   right: '-100px',
//   bottom: '150px',
//   transition: 'right 0.5s',
//   cursor: 'pointer',
//   backgroundColor: '#2B6CB0',
//   color: 'white',
//   fontSize: '20px',
//   padding: '10px',
//   borderColor: '#2B6CB0ed', // #2B6CB0
//   border: 'none',
//   fill: 'white',
//   focus: 'focus:outline-none',
// };

// const scrollButtonTransition = {
//   right: '20px',
// };