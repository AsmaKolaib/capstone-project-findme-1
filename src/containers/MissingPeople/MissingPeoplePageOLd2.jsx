import React, { useState, useEffect, useRef } from 'react';
import { CircleArrow as ScrollUpButton } from 'react-scroll-up-button';
import MissingPersonCard from '../../components/MissingPersonCard';
import { ReactComponent as MyIcon } from './repeat-grid-4.svg';
import Api from './api';
import './style.css';

const MissingPeoplePage = () => {
  const ITEMSTOSHOW = 8;

  const [data, setData] = useState([]); // maybe set the initial value later
  const [visible, setVisible] = useState(ITEMSTOSHOW);
  const [buttonText, setButtonText] = useState('⬇ Show More');
  const buttonTextRef = useRef('⬇ Show More');

  useEffect(() => {
    setData(() => Api);
  }, []);

  const showMoreItems = () => {
    if (visible > data.length) {
      // setButtonText(() => 'complete');
      buttonTextRef.current = 'Complete';
    }

    // setButtonText(() => '...loading');
    // buttonTextRef.current = '..loading';
    setVisible((prevValue) => prevValue + ITEMSTOSHOW);
    // buttonTextRef.current = '⬇ Show More';
    // setButtonText(() => '⬇ Show More');
  };

  const showMoreItemsClick = () => {
    // buttonTextRef.current = '..loading';
    setTimeout(() => {
      showMoreItems();
    }, 1000);
    clearTimeout();
    // buttonTextRef.current = '⬇ Show More';
  };

  return (
    <div>
      <MyIcon className="absolute" />
      <MyIcon className="absolute right-0 mt-64 " />

      <div className="flex flex-wrap ml-8">
        {data.length === 0 ? (
          <h2 className="flex-auto text-center">not found results</h2>
        ) : (
          data
            .slice(0, visible)
            .map((item, index) => (
              <MissingPersonCard key={item.id} cardInfo={item} i={index} />
            ))
        )}
      </div>
      {/*console.log(visible, data.length)*/}
      <div className=" ">
        {visible > data.length ? (
          <button
            // disabled
            className="appearance-none mx-auto my-10 py-2 px-4 align-center text-blue-500 border-solid border-2 border-blue-500 rounded-3xl hover:text-blue-600 hover:border-blue-600 block font-light "
            onClick={showMoreItemsClick}
            style={{ display: 'none' }}
          >
            <span className="">{buttonTextRef.current}</span>
          </button>
        ) : (
          <button
            className="focus:outline-none mx-auto my-10 py-2 px-4 align-center text-blue-500 border-solid border-2 border-blue-500 rounded-3xl hover:text-blue-600 hover:border-blue-600 block font-light "
            onClick={showMoreItemsClick}
          >
            <span className="">{buttonTextRef.current}</span>
          </button>
        )}
      </div>

      <ScrollUpButton
        style={scrollButtonStyle}
        onFocus={() => removeOutline}
        ContainerClassName="ScrollUpButton__Container"
        TransitionClassName="ScrollUpButton__Toggled"
      ></ScrollUpButton>
    </div>
  );
};

const removeOutline = () => {};

export default MissingPeoplePage;

const scrollButtonStyle = {
  position: 'fixed',
  right: '-100px',
  bottom: '150px',
  transition: 'right 0.5s',
  cursor: 'pointer',
  backgroundColor: '#2B6CB0',
  color: 'white',
  fontSize: '20px',
  padding: '10px',
  borderColor: '#2B6CB0ed', // #2B6CB0
  border: 'none',
  fill: 'white',
  focus: 'focus:outline-none',
};

// const scrollButtonTransition = {
//   right: '20px',
// };
