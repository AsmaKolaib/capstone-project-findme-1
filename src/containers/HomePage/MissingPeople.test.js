import React from 'react';
import ReactDOM from 'react-dom';
import { getByTestId, render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MissingPeople from './MissingPeople';

afterEach(cleanup);

it('render without crashes ', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MissingPeople />, div);
});

it('renders snapshot 1', () => {
  const tree = renderer
    .create(<MissingPeople missingPeople="Missing People" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
