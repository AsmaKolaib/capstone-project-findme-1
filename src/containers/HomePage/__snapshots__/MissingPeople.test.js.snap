// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1 1`] = `
<div
  className=""
>
  <h2
    className="font-normal tracking-wider text-4xl mb-10 mx-4"
  >
    Missing People
  </h2>
  <div
    className="flex flex-col md:flex-row lg:flex-row xl:flex-row sm:flex-col sm:items-center"
  >
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-6 py-2 font-bold md:pr-2 md:pl-3 md:py-1 lg:pl-6 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Found
          </span>
          <img
            alt="John Doe image"
            className="opacity-50 cursor-pointer"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "profile-svg.svg",
              }
            }
            width={300}
          />
        </div>
        <div
          className="px-6 py-4"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost sice: 6-6-2016
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-6 py-2 font-bold md:pr-2 md:pl-3 md:py-1 lg:pl-6 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Found
          </span>
          <img
            alt="Jane Doe image"
            className="opacity-50 cursor-pointer"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "profile-svg.svg",
              }
            }
            width={300}
          />
        </div>
        <div
          className="px-6 py-4"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            Jane Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost sice: 7-7-2017
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-6 py-2 font-bold md:pr-2 md:pl-3 md:py-1 lg:pl-6 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Lost
          </span>
          <img
            alt="Salem Saleh image"
            className="opacity-50 cursor-pointer"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "profile-svg.svg",
              }
            }
            width={300}
          />
        </div>
        <div
          className="px-6 py-4"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            Salem Saleh
          </div>
          <p
            className="text-base font-light"
          >
            Lost sice: 8-8-2018
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-6 py-2 font-bold md:pr-2 md:pl-3 md:py-1 lg:pl-6 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Lost
          </span>
          <img
            alt="Ahmed Khaled image"
            className="opacity-50 cursor-pointer"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "profile-svg.svg",
              }
            }
            width={300}
          />
        </div>
        <div
          className="px-6 py-4"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            Ahmed Khaled
          </div>
          <p
            className="text-base font-light"
          >
            Lost sice: 9-9-2019
          </p>
        </div>
      </div>
    </div>
  </div>
  <p
    className="mr-10 my-10 text-right text-blue-500"
  >
    <a
      href="#"
    >
      <span
        className=""
      >
        More Missing People â†’
      </span>
    </a>
  </p>
</div>
`;
